pkg/endpoint/bpf.go

```diff
@@ -172,6 +172,45 @@ func (e *Endpoint) writeHeaderfile(prefix string) error {
 	return f.CloseAtomicallyReplace()
 }
 
+// proxyPolicy implements policy.ProxyPolicy interface, and passes most of the calls
+// to policy.L4Filter, but re-implements GetPort() to return the resolved named port,
+// instead of returning a 0 port number.
+type proxyPolicy struct {
+	l4   *policy.L4Filter
+	port uint16
+}
+
+// newProxyPolicy returns a new instance of proxyPolicy by value
+func (e *Endpoint) newProxyPolicy(l4 *policy.L4Filter, port uint16) proxyPolicy {
+	return proxyPolicy{l4: l4, port: port}
+}
+
+// CopyL7RulesPerEndpoint returns a copy of L7 rules for the proxy
+func (p *proxyPolicy) CopyL7RulesPerEndpoint() policy.L7DataMap {
+	return p.l4.CopyL7RulesPerEndpoint()
+}
+
+// GetL7Parser returns the parser type of the proxy policy
+func (p *proxyPolicy) GetL7Parser() policy.L7ParserType {
+	return p.l4.L7Parser
+}
+
+// GetIngress returns 'true' if the proxy policy is for ingress, false if it is for egress
+func (p *proxyPolicy) GetIngress() bool {
+	return p.l4.Ingress
+}
+
+// GetPort returns the destination port number on which the proxy policy applies
+// This version properly returns the port resolved from a named port, if any.
+func (p *proxyPolicy) GetPort() uint16 {
+	return p.port
+}
+
+// GetListener returns the listener name referenced by the policy, if any
+func (p *proxyPolicy) GetListener() string {
+	return p.l4.GetListener()
+}
+
```

Comment:
We can simplify a little bit by only implementing the methods that are different.
```suggestion
type proxyPolicy struct {
	*policy.L4Filter
	port uint16
}

// newProxyPolicy returns a new instance of proxyPolicy by value
func (e *Endpoint) newProxyPolicy(l4 *policy.L4Filter, port uint16) proxyPolicy {
	return proxyPolicy{L4Filter: l4, port: port}
}

// GetPort returns the destination port number on which the proxy policy applies
// This version properly returns the port resolved from a named port, if any.
func (p *proxyPolicy) GetPort() uint16 {
	return p.port
}
```

Reply:
Done, thanks!